pkg/api/validation/validation.go

```diff
@@ -613,6 +613,9 @@ func validateHostPathVolumeSource(hostPath *api.HostPathVolumeSource, fldPath *f
 	if len(hostPath.Path) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("path"), ""))
 	}
+
+	typeErrs := validateHostPathType(*(hostPath.Type), fldPath.Child("type"))
```

Comment:
pass the pointer or test for nil.  Interestingly, if it is nil at this point, it's an internal error - defaulting should have caught it.